lib/models/artist.dart

```diff
@@ -0,0 +1,81 @@
+import '../util/int_util.dart';
+
+class Artist {
+  final ArtistExternalUrls externalUrls;
+  final ArtistFollowers followers;
+  final List<String> genres;
+  final String href;
+  final String id;
+  final List<ArtistImages> images;
+  final String name;
+  final int popularity;
+  final String uri;
+
+  Artist({
+    required this.externalUrls,
+    required this.followers,
+    required this.genres,
+    required this.href,
+    required this.id,
+    required this.images,
+    required this.name,
+    required this.popularity,
+    required this.uri,
+  });
+
+  factory Artist.fromJson(Map<String, dynamic>? json) {
```

Comment:
O JsonSerialize cuida das conversões e as tiragens, caso não opta pelo Freezed, sugiro usar o https://docs.flutter.dev/development/data-and-backend/json

Reply:
Eu entendo que as libs ajudam a diminuir o boilerplate, mas eu ainda prefiro nao utilizar nenhuma package externa para este tipo de operacao. Ja tive problemas de resiliencia, como por exemplo em tratativas de valores que poderiam ser facilmente convertidos (como um valor int que vem como uma string: "valor": "20").

Reply:
> Eu entendo que as libs ajudam a diminuir o boilerplate, mas eu ainda prefiro nao utilizar nenhuma package externa para este tipo de operacao. Ja tive problemas de resiliencia, como por exemplo em tratativas de valores que poderiam ser facilmente convertidos (como um valor int que vem como uma string: "valor": "20").

Sim sim! Temos esse efeito colateral.

Reply:
Exatamente!!! rsrsrs!!! Ja tive app quebrada por conta de tanta besteira!! E o dificil era convencer os outros que: "20", 20.0 e 20 sao 3 coisas diferentes! Enfim, por conta dessas e outras eu prefiro ter um trabalho um pouco maior (co-pilot me salva na hora de digitar) e escrever boilerplate, mas tendo o controle de tudo!! 

Note a funcao "IntUtil.parseInt" que coloquei no from json! Ela trata "20", 20.0 e 20 como 20. E segue o jogo!